autogpt_platform/frontend/src/components/auth/GoogleOAuthButton.tsx

```diff
@@ -0,0 +1,33 @@
+import { useState } from "react";
+import { FaGoogle, FaSpinner } from "react-icons/fa";
+import { Button } from "../ui/button";
+
+interface GoogleOAuthButtonProps {
+  onClick: () => void;
+  isLoading?: boolean;
+  disabled?: boolean;
+}
+
+export default function GoogleOAuthButton({
+  onClick,
+  isLoading = false,
+  disabled = false,
+}: GoogleOAuthButtonProps) {
+  return (
+    <Button
+      type="button"
+      className="w-full border bg-zinc-700 py-2 text-white disabled:opacity-50"
+      disabled={isLoading || disabled}
+      onClick={onClick}
+    >
+      {isLoading ? (
+        <FaSpinner className="mr-2 h-4 w-4 animate-spin" />
+      ) : (
+        <FaGoogle className="mr-2 h-4 w-4" />
+      )}
+      <span className="text-sm font-medium">
+        {isLoading ? "Signing in..." : "Continue with Google"}
+      </span>
+    </Button>
+  );
+}
```

Comment:
We should be encouraging people to SSO with Google ( _or other providers_ ) rather than use passwords. 
Passwords come with security risks and maintenance overhead 💆🏽 .  Hence why I made the Google login button primary in the form.

This buttons need to adhere to our design-system, but doing that would come as a separate piece of work. For now I made it look similar to the existing login button ( which has `shadcn` default styling )